src/ResourceManager/Profile/Commands.Profile/Utilities/EnvironmentHelper.cs

```diff
@@ -0,0 +1,83 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.Profile.Utilities
+{
+    using Microsoft.Azure.Commands.Profile.Models;
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Http;
+    using System.Threading.Tasks;
+    using Hyak.Common;
+    using Newtonsoft.Json;
+
+    internal static class EnvironmentHelper
+    {
+        /// <summary>
+        /// Retrieves the domain.
+        /// </summary>
+        /// <param name="portalEndpoint">The portal endpoint.</param>
+        /// <returns>Domain</returns>
+        internal static string RetrieveDomain(string portalEndpoint)
+        {
+            // Todo: Revisit this when azure metadata endpoint supports keyvault suffix and storage endpoints
+            // Example format:: portal endpoint: "management.azure.com"; returns: "azure.com"
+            string domainHost = new Uri(portalEndpoint).Host;
+            return domainHost.Replace(domainHost.Split('.')[0], "").TrimEnd('/').TrimStart('.');
```

Comment:
You should not need the ```TrimEnd('/')``` call, as the Host fiels will not contain any part of the path